### ISSUE TITLE
Deprecate default heading tag from `Subhead` and `Blankslate`

### ISSUE THREAD

**Issue by khiga8** [primer/view_components#630]
Been on my radar but filing issue for tracking as it was brought up for [discussion](https://github.com/github/ui-platform/discussions/492).

`Subhead` currently defaults to `:div` and `Blankslate` currently defaults `title_tag` to `:h3`. 
We should always require the consumer to pass in a heading tag instead of having a default because:

1. There's no guarantee the default or fixed heading level is correct in all places the component is used. 
2. In a view component implementation, when looking at the code, `Primer::SomethingComponent.new(tag: :h2)` provides heading information that developers should be aware of when updating the page vs `Primer::SomethingComponent.new` which obscures this information.
3. Requiring the heading tag to be passed in helps developers be more cognizant of heading levels and avoid skip-level heading issues. 

Since we have 500+ uses for both these components in dotcom, changing this API requires some consideration for how to update those uses.

**Comment by khiga8** [comment]
@alliethu @manuelpuyol 

Related to API & Accessibility for https://github.com/github/primer/issues/317
